### ISSUE TITLE
Support for ASCII input.

### ISSUE THREAD

**Issue by lvella** [reu/cpf-rs#5]
We often get CPF values from places where it was not properly validated as UTF-8. So turning `&[u8]` into `&str` and then into `Cpf` is kind of a waste of the first step, since all valid UTF-8 CPFs are strictly ASCII. Thus, it would be nice to have a parse interface from ASCII directly.

**Comment by ruanpetterson** [comment]
> We often get CPF values from places where it was not properly validated as UTF-8. So turning `&[u8]` into `&str` and then into `Cpf` is kind of a waste of the first step, since all valid UTF-8 CPFs are strictly ASCII. Thus, it would be nice to have a parse interface from ASCII directly.

Doesn't the current [`TryFrom<&[u8]>`](https://github.com/reu/cpf-rs/blob/793c51094e69b955c1352816f77cbe22b3fb099b/src/lib.rs#L275-L281) implementation already cover this case?

**Comment by lvella** [comment]
> Doesn't the current [`TryFrom<&[u8]>`](https://github.com/reu/cpf-rs/blob/793c51094e69b955c1352816f77cbe22b3fb099b/src/lib.rs#L275-L281) implementation already cover this case?

No, because `TryFrom<&[u8]>` treats each `u8` as the numeric value of the digit, not as the ASCII character.

**Comment by lvella** [comment]
BTW, that one is broken, see issue #6.

**Comment by ruanpetterson** [comment]
Given that the caller guarantees the input is ASCII, wouldn't it be possible for the caller to directly convert the `&[u8]` to a `&str` using [`str::from_utf8_unchecked`](https://doc.rust-lang.org/core/str/fn.from_utf8_unchecked.html)?

**Comment by lvella** [comment]
The idea is that the caller doesn't have to guarantee it.

The user expects an untrusted &`[u8]` to be an ASCII CPF, and I am proposing a function that validates it as a CPF directly, without the step of validating it as UTF-8.

**Comment by reu** [comment]
The decision up to this point it to treat `u8` as a numeric digit and not as a byte/char. That might not be the most "correct" assumption, but that's how it is now...

BTW this would be a great use case for https://doc.rust-lang.org/std/ascii/enum.Char.html, but it still nightly only.

**Comment by reu** [comment]
FYI https://github.com/reu/cpf-rs/pull/8
